File: scrapper/src/match/match.py
from dataclasses import dataclass
from typing import Any, List
from team.models import Team

@dataclass
class EventType:
    id: str
    text: str

@dataclass
class Event:
    type: EventType
    text: str
    clock_value: int
    team_id: int
    participant_name: str

class Competitor:
    home_away: str
    winner: bool
    score: int
    team: Team
    roster: str

class Match:
    id: int
    venue: str
    date: str
    note: str
    completed: bool
    status_name: str
    competition_name: str
    home_competitor: Competitor
    away_competitor: Competitor
    events: List[Event]

File: football/internal/match/match.go
package match

import "github.com/modasby/futeboxd-api/services/football/internal/team"

type Competitor struct {
	HomeAway string    `json:"homeAway"`
	Winner   bool      `json:"winner"`
	Score    int32     `json:"score"`
	Team     team.Team `json:"team"`
}

type Match struct {
	ID              int64      `json:"id"`
	Venue           string     `json:"venue"`
	Date            string     `json:"date"`
	Note            string     `json:"note"`
	Completed       bool       `json:"completed"`
	StatusName      string     `json:"status_name"`
	CompetitionName string     `json:"competition_name"`
	HomeCompetitor  Competitor `json:"home_competitor"`
	AwayCompetitor  Competitor `json:"away_competitor"`
}

func NewMatch(
	ID int64,
	venue string,
	date string,
	note string,
	homeCompetitor Competitor,
	awayCompetitor Competitor,
	completed bool,
	statusName string,
	competitionName string,
) *Match {
	return &Match{
		ID:              ID,
		Venue:           venue,
		Date:            date,
		Note:            note,
		HomeCompetitor:  homeCompetitor,
		AwayCompetitor:  awayCompetitor,
		Completed:       completed,
		StatusName:      statusName,
		CompetitionName: competitionName,
	}
}

func NewCompetitor(homeAway string, winner bool, score int32, team team.Team) *Competitor {
	return &Competitor{
		HomeAway: homeAway, Winner: winner, Score: score, Team: team,
	}
}
